src/interfaces/simulation.ts

```diff
@@ -37,85 +60,40 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     toVault: Address,
     options: SimulationOptions = {}
   ): Promise<TransactionOutcome> {
+    const depositArgs = { from, sellToken, amount, toVault, options };
+
     const signer = this.ctx.provider.write.getSigner(from);
-    const zapProtocol = PickleJars.includes(toVault) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-    const vaultContract =
-      zapProtocol === ZapProtocol.PICKLE
-        ? new PickleJarContract(toVault, signer)
-        : new YearnVaultContract(toVault, signer);
-
-    const underlyingToken = await vaultContract.token().catch(() => {
-      throw new EthersError("failed to fetch token", EthersError.FAIL_TOKEN_FETCH);
-    });
-    const isZapping = underlyingToken !== sellToken;
-    let forkId: string | undefined;
-    let simulateDeposit: (save: boolean) => Promise<TransactionOutcome>;
 
-    if (isZapping) {
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
+    const vaultContract = this.getVaultContract({ toVault, signer });
 
-      let needsApproving: boolean;
+    const vaultMetadata = await this.yearn.services.meta.vault(toVault);
```

Comment:
Should fetch vault metadata from `vault.getDynamic`

Reply:
Good catch, refactored with https://github.com/yearn/yearn-sdk/pull/280/commits/b62b06bda5904632f7592a413039ee2e0a08a305